storage/storage.go

```diff
@@ -90,55 +90,132 @@ func (storage *Storage) getAllChunks(context context.Context, buildId string) ([
 	return buildChunks, nil
 }
 
-func (storage *Storage) GetTestLogLines(context context.Context, buildId string, testId string) (LogIterator, error) {
+func (storage *Storage) getBuildAndTestChunks(context context.Context, buildId string) ([]LogChunkInfo, []LogChunkInfo, error) {
 	chunks, err := storage.getAllChunks(context, buildId)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
+	}
+
+	buildChunks := []LogChunkInfo{}
+	for i := 0; i < len(chunks); i++ {
+		if chunks[i].TestID == "" {
+			buildChunks = append(buildChunks, chunks[i])
+		}
 	}
 
 	testChunks := []LogChunkInfo{}
 	for i := 0; i < len(chunks); i++ {
 		// Find our test id
-		if chunks[i].TestID == testId {
+		if chunks[i].TestID != "" {
 			testChunks = append(testChunks, chunks[i])
 		}
 	}
-	sort.Slice(testChunks, func(i, j int) bool {
-		return testChunks[i].Start.Before(testChunks[j].Start)
-	})
+	return buildChunks, testChunks, nil
+}
 
-	var latestTime = testChunks[len(testChunks)-1].End
-	for _, chunk := range testChunks {
+func getLatestTime(chunks []LogChunkInfo) time.Time {
+	var latestTime = chunks[len(chunks)-1].End
+	for _, chunk := range chunks {
 		if chunk.End.After(latestTime) {
 			latestTime = chunk.End
 		}
 	}
+	return latestTime
+}
 
-	testTimeRange := TimeRange{
-		StartAt: testChunks[0].Start,
-		EndAt:   testChunks[len(testChunks)-1].End,
+func sortByStartTime(chunks []LogChunkInfo) {
+	sort.Slice(chunks, func(i, j int) bool {
+		return chunks[i].Start.Before(chunks[j].Start)
+	})
+}
+
+func minTime(first time.Time, second time.Time) time.Time {
+	if first.Before(second) {
+		return first
+	} else {
+		return second
 	}
+}
 
-	testChunkIterator := NewBatchedLogIterator(storage.bucket, testChunks, 4, testTimeRange)
+func maxTime(first time.Time, second time.Time) time.Time {
+	if first.Before(second) {
+		return second
+	} else {
+		return first
+	}
+}
 
-	buildChunks := []LogChunkInfo{}
+func (storage *Storage) GetAllLogLines(context context.Context, buildId string) (LogIterator, error) {
+	buildChunks, testChunks, err := storage.getBuildAndTestChunks(context, buildId)
+	if err != nil {
+		return nil, err
+	}
+
+	sortByStartTime(buildChunks)
+	sortByStartTime(testChunks)
+
+	timeRange := TimeRange{
+		StartAt: minTime(buildChunks[0].Start, testChunks[0].Start),
+		EndAt:   maxTime(getLatestTime(buildChunks), getLatestTime(testChunks)),
```

Comment:
it would be nice if TimeRange had an option to not specify a min and/or max time. Like we don't really need a min or max time here: it's just to signal to the iterator not to filter anything out.

Reply:
I solved this by adding TimeRangeMin and TimeRangeMax. I looked into making it optional, but that makes a lot of the logic hard and I don't think we need to represent times outside of the range 1970 - 2200